src/commands/generate/fromTemplate.ts

```diff
@@ -99,14 +122,21 @@ export default class Template extends Command {
       mapBaseUrlToFolder: parsedFlags.mapBaseUrlToFolder,
       disabledHooks: parsedFlags.disableHooks,
     };
+    const asyncapiInput = (await load(asyncapi)) || (await load());
 
     const watchTemplate = flags['watch'];
     const genOption: any = {};
     if (flags['map-base-url']) {
       genOption.resolve = {resolve: this.getMapBaseUrlToFolderResolver(parsedFlags.mapBaseUrlToFolder)};
     }
 
-    await this.generate(asyncapi, template, output, options, genOption);    
+    if (asyncapiInput.isAsyncAPI3()) {
+      const v3IssueLink = verifyTemplateSupportForV3(template);
+      if (v3IssueLink !== undefined) {
+        this.error(`${template} template does not support AsyncAPI v3 documents, please checkout ${v3IssueLink}`);
+      }
+    }
```

Comment:
what happens when I pass in a custom template that does not support v3?

Reply:
Nothing, we cant expect the unexpected 🙂 

Custom templates can easily be added to the list of templates that don't support v3, so all about contributing 😄

Reply:
> Nothing, we cant expect the unexpected 🙂

Yeah, that make sense. 

I was thinking that since templates have a config in the `package.json` they might have a parameter that could help us determine which spec versions this template supports, but now that I have checked I don't think there is anything like that.